ProgressBar y control de errores

TheDiegoDev opened the pull request:
He añadido un progressBar y un control de errores

TheDiegoDev committed:
ProgressBar y control de errores

He añadido un progressBar y un control de errores

jgorozco commented on app/src/main/java/guinea/diego/myrecycleview/MainActivity.kt:
```diff
@@ -27,13 +29,15 @@ class MainActivity : AppCompatActivity() {
 
             override fun onFailure(call: Call<Characters>, t: Throwable) {
-                println("deu ruim" + t.message)
+                progressBar.visibility = View.INVISIBLE
```
te recomiendo sacar este tipo de cosas a funciones. 

Imagina un Mainview con funciones del tipo:
progressBarStatus(showProgress:Boolean)

inicialmente seguramente sea como has puesto aqui, pero eventualmente querremos meterle animaciones, cerrar o abrir cosas... además permite viendo las funciones de la actividad saber que haces

jgorozco commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -9,8 +9,32 @@
+      android:layout_height="wrap_content"
+      app:layout_constraintBottom_toBottomOf="parent"
+      app:layout_constraintEnd_toEndOf="parent"
+      app:layout_constraintHorizontal_bias="0.498"
```
esto creo que no es necesario, al final siempre está centrado en el centro, es normal cuando lo haces con el editor, pero es bueno revisar lo que genera en el xml

TheDiegoDev committed:
Cambio de pantalla con recyclerview y optimizacion del codigo

Se realiza un cambio de pantalla al clickar un personaje en el recyclerview y he aplicado los consejos sobre el comentario del PR anterior

TheDiegoDev committed:
optimizacion del codigo
